### DISCUSSION TITLE
Nested data error with re-usable component / single-use component combination

### DISCUSSION THREAD

**Discussion by stuartcusackie** [alpinejs/alpine#2781] [7. Bugs (archived)] [answered]
I'm not quite sure what's going on so my apologies if this is not a bug.

I have a single-use component like so:
`<a x-data="{hover:false}" x-on:mouseover="hover=true" x-on:mouseover.away="hover=false">`

And within it I have a reusable component:
`<div x-data="reusable"  x-bind:class="{ 'brightness-125': hover }">`

I am seeing this error:
`VM12165:3 Uncaught (in promise) ReferenceError: hover is not defined`

Is there an issue when accessing nested properties in a single-use / re-usable component combination?

**Comment by SimoTod** [comment]
It works. there's probably an issue with your code.
As a separate note, depending on how you structure your html, `mouseover.away` is not going to do what you need. Probably you want `mouseout`.

**Comment by stuartcusackie** [comment, marked as answer]
Thanks, problem solved. It appears that having an element in between the two components was causing the problem:

```
<a x-data="{hover:false}" x-on:mouseover="hover=true" x-on:mouseover.away="hover=false">
   <picture x-data="reusable">
     <img x-bind:class="{ 'brightness-125': hover }" />
   </picture>
</a>
```

When I moved the bind to the picture element instead, everything worked fine:

```
<a x-data="{hover:false}" x-on:mouseover="hover=true" x-on:mouseover.away="hover=false">
   <picture x-data="reusable" x-bind:class="{ 'brightness-125': hover }">
     <img  />
   </picture>
</a>
```
